OpenRA.Game/Map/Map.cs

```diff
@@ -1072,16 +1039,15 @@ public byte GetTerrainIndex(CPos cell)
 			if (terrainIndex == InvalidCachedTerrainIndex)
 			{
 				var custom = CustomTerrain[uv];
-				terrainIndex = cachedTerrainIndexes[uv] =
-					custom != byte.MaxValue ? custom : Rules.TileSet.GetTerrainIndex(Tiles[uv]);
+				terrainIndex = cachedTerrainIndexes[uv] = custom != byte.MaxValue ? custom : Rules.TerrainInfo.GetTerrainInfo(Tiles[uv]).TerrainType;
```

Comment:
I had an unexpected null reference exceptions here during the [blank shellmap](https://github.com/OpenHV/OpenHV/tree/master/mods/hv/maps/blank-map) load so this might crash in new and unexpected ways, which I debugged down to this
```suggestion
				var custom = CustomTerrain[uv];
				var tile = Tiles[uv];
				var terrainInfo = Rules.TerrainInfo.GetTerrainInfo(tile);
				if (terrainInfo == null)
					terrainIndex = 0;
				else
				{
					var terrainType = terrainInfo.TerrainType;
					terrainIndex = cachedTerrainIndexes[uv] = custom != byte.MaxValue ? custom : terrainType;
				}
```

Reply:
I will instead file a PR to improve the handling of maps that reference undefined tiles. We can then remove all of these awkward null checks from the code.

Reply:
The weird thing is that it referenced tile 255 and 0 which are both defined. I also saw nothing too strange in the hex editor. Redoing the map with just tile 0 fixed it. https://github.com/OpenHV/OpenHV/commit/9062973b2791dfd09dff47761c80c7cabda1077f

Reply:
Actually this happens when referencing non-existing tile IDs. I have some holes after removing duplicates in my tileset. Maybe just add a friendlier error message that references the tile ID.

Reply:
Fixed in #18927.